build/cmake/BoostMacros.cmake

```diff
@@ -19,6 +19,18 @@
 
 set(BOOST_MINREV 1.53)
 
+macro(ADD_BOOST_OPTIONS)
+  if (MSVC)
+    add_definitions(/DBOOST_ALL_DYN_LINK)
```

Comment:
We may not need this at all - if someone defines `Boost_USE_STATIC_LIBS` as part of the cmake command line then FindBoost might do the right thing for us.  As it is if someone does define `Boost_USE_STATIC_LIBS` this macro is wrong as these things should not be defined.

Reply:
fixed